discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java

```diff
@@ -307,8 +326,52 @@ private void updateClusterAddressMap(Jedis jedis, String redisRegistryKey, Strin
             }
         } while (!cursor.equals(ScanParams.SCAN_POINTER_START));
 
-        if (!newAddressSet.equals(CLUSTER_ADDRESS_MAP.get(clusterName))) {
-            CLUSTER_ADDRESS_MAP.put(clusterName, newAddressSet);
+        if (CollectionUtils.isNotEmpty(newAddressSet) && !newAddressSet.equals(CLUSTER_ADDRESS_MAP.get(redisMessage.getCluster()))) {
+            CLUSTER_ADDRESS_MAP.put(redisMessage.getCluster(), newAddressSet);
+        }
+    }
+
+    /**
+     *
+     * handle redis message
+     *
+     */
+    public class RedisMessageHandler implements Consumer<RedisMessage> {
+
+        /***
+         *
+         * @param redisMessage
+         */
+        @Override
+        public void accept(RedisMessage redisMessage) {
+            reentrantLock.lock();
+            try {
+                if (redisMessage.getMessageType() == MessageType.SUBSCRIBE) {
```

Comment:
为什么messagetype不是add或者remove？而是scan和SUBSCRIBE？我希望是scan那部分找到的列表能跟当前的cluster中的集合做对比，将需要add或者remove的事件发出，而不是在这里面写ifelse，结果还是各玩各的。我希望这里只是一个动作触发，要么是remove，要么是add
Why is messagetype not add or remove? But scan and SUBSCRIBE? I hope that the list found in the scan part can be compared with the collection in the current cluster, and the events that need to be added or removed will be sent out, instead of writing ifelse in it, the result is still different. I hope this is just an action trigger, either remove or add.